abdul-moize/wheater-man
# Add Flag `-e`, `-a` & `-c`. All Tasks of weatherman

This PR adds all the tasks of weatherman to main branch.
- Task 1 (`-e`) finds and displays the `Highest & Lowest Temperatures` and `Max Humidity` for a given `year`
- Task 2 ('-a') calculates averages of `Highest & Lowest Temperatures` and `Mean Humidity` for a given `month`
- Task 3 (`-c`), for each `day` of a given `month` displays  `Highest & Lowest Temperatures` on two lines
- Task 4 is the ability to run multiple tasks/reports at once
- Task 5 (`-c`), for each `day` of a given `month` displays  `Highest & Lowest Temperatures` on one line

To test this PR, follow these steps:
1. pull this branch
2. run `pip install -r requirements.txt`
3. run `weatherman.py` with below parameters:
-  `-a 2006/6` should return
![image](https://user-images.githubusercontent.com/88365652/130785998-8cf78801-48fb-4976-bd01-595260c87ba5.png)

- `-c 2006/6` should return
![image](https://user-images.githubusercontent.com/88365652/130788539-907aa336-9d4f-4157-9709-3c41452296b0.png)
![image](https://user-images.githubusercontent.com/88365652/130788613-a7ac1f54-9963-4ad4-8567-8e6525799627.png)

- `-e 2005` should return
![image](https://user-images.githubusercontent.com/88365652/130788774-87c368e3-0ece-4a29-96b8-7f7f5bd20ab7.png)

- `-a 2005/4 -e 2005 -c 2006/2`
![image](https://user-images.githubusercontent.com/88365652/130788078-ec2e8883-c77d-42ce-aa25-2dfb6b88461a.png)
![image](https://user-images.githubusercontent.com/88365652/130788230-bb3ba65a-773d-439b-b521-7c188aec2d47.png)

Please try `different values` of `parameters` and let me know if there are any issues
